README.txt
Extension: t3notes

- lib/
Offers a libary for authentication to a Lotus Notes Server

- sv1/
Offers a auth service for the authentication of a fe_user or general website user to Lotus Notes Server through the libary above

- ext_autoload.php
Makes sure that the class in lib/class.tx_t3notes_auth.php is loaded once only


Use 'options.disableNotesTab' as UserTS option for disabling the notes tab


ChangeLog
2011-08-25 Sonja Scholz <sonja.scholz@typo3.org>
	* Version 0.3.0
	* added documentation

2011-08-05 Nils Blattner <nb@cabag.ch>
	* Version 0.2.9
	* t3notes/ext_conf_template.txt - fixed type "bool" to "boolean"

2011-08-03 Nils Blattner <nb@cabag.ch>
	* Version 0.2.8
	* t3notes/sv1/class.tx_t3notes_sv1.php - fixed a few bugs when logging in via the cookie
	* t3notes/sv1/class.tx_t3notes_sv1.php - added possibility to define permanent cookie login via ext conf
	* t3notes/ext_conf_template.txt - added key for the permanent cookie login

2011-08-02 Nils Blattner <nb@cabag.ch>
	* Version 0.2.7
	* changed t3notes/sv1/class.tx_t3notes_sv1.php - added possibility to store the notes information in a cookie and autologin with it.

2011_03_31 Sonja Scholz <ss@cabag.ch>
	* Version 0.2.6
	* changed class.tx_t3notes_browselinkshooks.php - added condition to decide if the notes tab content should be added to prevent double content

2011-03-25 Sonja Scholz <ss@cabag.ch>
	* Version 0.2.5
	* changed sv/sv1 php - changed encoding of NTLM key because in TYPO3 4.5.2 the post is urldecoded automatically

2011-03-24 Sonja Scholz  <ss@cabag.ch>
	* Version 0.2.4
	* changed hooks/class.tx_t3notes_ts_links_rte_process.php - added function postProcessHref
	* added version folder for xclasses in ext root
	* added new xclass versions in version folder 4.5.2

2010-12-08 Sonja Scholz  <ss@cabag.ch>
	* Version 0.2.3
	* changed auth class - write the detail informations to devlog on error in authenticate function
	* changed auth class - changed getRequestCOOKIE() - return false if no cookie was found
	* changed auth class - changed request() - return false if no cookie was received by other function
	* changed auth class - don't throw an exception if the lotus notes server curl request failed, be silent

2010-08-19 Sonja Scholz  <ss@cabag.ch>
	* Version 0.2.2

2010-08-19 Sonja Scholz  <ss@cabag.ch>
	* Version 0.2.1
	* cookie domain can be set in the extension manager now - if empty cookie domain from lotus notes will be used

2010-08-18 Sonja Scholz  <ss@cabag.ch>
	* Version 0.2.0
	* fixed bug in mcrypt functionaltity

2010-08-17 Sonja Scholz  <ss@cabag.ch>
	* Version 0.1.9
	* changed NTLM login functionaltity in tx_t3notes_sv1 - read the username from a URL param instead of reading from a PHP session

2010-07-12 Sonja Scholz  <ss@cabag.ch>
	* Version 0.1.8
	* fixed bug in auth class - don't write any log to devLog if the localconf setting issn't enabled

2010-06-25 Sonja Scholz  <ss@cabag.ch>
	* Version 0.1.7
	* added new option autoLoginWaitSeconds to ext_conf_template to configure the auto-login feature
	* added userFunc checkT3NotesAutoLogin() to auth class to call from page.5 Typoscript before rendering the FE content

2010-06-08 Sonja Scholz  <ss@cabag.ch>
	* Version 0.1.6
	* add new function processCurlAdditionalPostParams to cabag_phpproxy hook

2010-06-07 Sonja Scholz  <ss@cabag.ch>
	* Version 0.1.5
	* added new extension manager option curlTimeout

2010-06-03 Sonja Scholz  <ss@cabag.ch>
	* Version 0.1.4
	* fixed setcookie bug in auth class - don't urlencode the cookie
	* fixed parse cookie bug in auth class - don't just split by = maybe there is a = in the value
	* set cookie domain fix because of temporary notes bug

2010-05-14 Jonas Duebi <jd@cabag.ch>
	* Version 0.1.3
	* enabled NTLM Login over sesson setting
	* NTLM $_SERVER['REQUEST_USER'] won't work
	* User name is put into the $_SESSION for further logins...
	* extended devLog calls / fixed devLog calls

2010-05-07 Dimitri Koenig  <dk@cabag.ch>
	* Version 0.1.2
	* fixed notes:/// redirect

2010-05-05 Dimitri Koenig  <dk@cabag.ch>
	* Version 0.1.1
	* added 'options.disableNotesTab' option for userts for disabling the notes tab for individual groups/users

2010-05-04 Sonja Scholz  <ss@cabag.ch>
	* Version 0.1.0
	* fixed some bugs in NTLM login
	* added function checkNTLMLogin() to check the NTLM login at a separate place
	* added function fetchUserRecord() 

2010-05-03 Sonja Scholz  <ss@cabag.ch>
	* Version 0.0.9
	* added functionality for NTLM login

2010-04-28 Sonja Scholz  <ss@cabag.ch>
	* Version 0.0.8
	* changed auth class - added some devlog calls to request function

2010-04-26 Dimitri Koenig  <dk@cabag.ch>
	* Version 0.0.7
	* finished notes:// tabs in rte

2010-04-26 Sonja Scholz  <ss@cabag.ch>
	* Version 0.0.6
	* fixed bug in sv1 - don't search for the lotus notes password in the t3 db
	* fixed bug in sv1 - don't save the lotus notes password while creating a new fe_user, save a random password instead

2010-04-12 Dimitri König  <dk@cabag.ch>
	* Version 0.0.5
	* added 'notes://' meta redirect to external url sites

2010-04-06 Sonja Scholz  <ss@cabag.ch>
	* Version 0.0.4
	* 

2010-03-30 Sonja Scholz  <ss@cabag.ch>
	* Version 0.0.3
	* fixed several bugs in authentication service

2010-03-30 Dimitri König  <dk@cabag.ch>
	* Version 0.0.2
	* added cag_linkchecker hook for fetching 'notes://' links

2010-03-25 Sonja Scholz  <ss@cabag.ch>
	* Version 0.0.1
	* added tx_t3notes_auth class
	* added ext_autoload.php for tx_t3notes_auth to don't have to add require once ot all scripts
	* added tx_t3notes_sv1 service class for getUserFE and authUserFE


ext_emconf.php
<?php

########################################################################
# Extension Manager/Repository config file for ext "t3notes".
#
# Auto generated 25-08-2011 11:03
#
# Manual updates:
# Only the data in the array - everything else is removed by next
# writing. "version" and "dependencies" must not be touched!
########################################################################

$EM_CONF[$_EXTKEY] = array(
	'title' => 'Lotus Notes Basis',
	'description' => 'Manages the authentification of fe_users and T3 requests to a Lotus Notes server.',
	'category' => 'services',
	'author' => 'Sonja Scholz',
	'author_email' => 'sonja.scholz@typo3.org',
	'shy' => '',
	'dependencies' => '',
	'conflicts' => '',
	'priority' => '',
	'module' => '',
	'state' => 'stable',
	'internal' => '',
	'uploadfolder' => 0,
	'createDirs' => '',
	'modify_tables' => '',
	'clearCacheOnLoad' => 1,
	'lockType' => '',
	'author_company' => '',
	'version' => '0.3.0',
	'constraints' => array(
		'depends' => array(
		),
		'conflicts' => array(
		),
		'suggests' => array(
		),
	),
	'_md5_values_when_last_written' => 'a:21:{s:9:"ChangeLog";s:4:"63b6";s:10:"README.txt";s:4:"8ac2";s:16:"ext_autoload.php";s:4:"39a8";s:21:"ext_conf_template.txt";s:4:"ea0b";s:12:"ext_icon.gif";s:4:"6262";s:17:"ext_localconf.php";s:4:"19c1";s:14:"ext_tables.php";s:4:"3f5b";s:13:"locallang.xml";s:4:"e77a";s:14:"doc/manual.sxw";s:4:"ace4";s:19:"doc/wizard_form.dat";s:4:"4235";s:20:"doc/wizard_form.html";s:4:"e361";s:43:"hooks/class.tx_t3notes_browselinkshooks.php";s:4:"f690";s:40:"hooks/class.tx_t3notes_cabagphpproxy.php";s:4:"752c";s:47:"hooks/class.tx_t3notes_ts_links_rte_process.php";s:4:"caa4";s:47:"lib/class.tx_caglinkchecker_checknoteslinks.php";s:4:"28cf";s:29:"lib/class.tx_t3notes_auth.php";s:4:"deea";s:28:"sv1/class.tx_t3notes_sv1.php";s:4:"8d55";s:44:"typo3_versions/4.3.2/class.ux_t3lib_page.php";s:4:"9048";s:69:"typo3_versions/4.3.2/class.ux_tx_templavoila_mod1_specialdoktypes.php";s:4:"b740";s:44:"typo3_versions/4.5.2/class.ux_t3lib_page.php";s:4:"2fb4";s:69:"typo3_versions/4.5.2/class.ux_tx_templavoila_mod1_specialdoktypes.php";s:4:"6d87";}',
	'suggests' => array(
	),
);

?>